airflow/www/views.py

```diff
@@ -3622,8 +3622,10 @@ def grid_data(self):
             if run_state:
                 query = query.filter(DagRun.state == run_state)
 
-            dag_runs = query.order_by(DagRun.execution_date.desc()).limit(num_runs).all()
+            ordering = (DagRun.__table__.columns[name].desc() for name in dag.timetable.run_ordering)
```

Comment:
Should we have the UI read `ordering` in order to know when to use `data_interval_end` as a label instead of `data_interval_start`?

Reply:
Ah, yes, we should. I’ll add a field in the response JSON and leave the UI part to you.

Reply:
Yeah, that'll be good. But let's do that as a subsequent PR since it breaks a lot of tests and needs a bunch of UI changes.

Reply:
Reverted the addition.